src/main/scala/asciibot/parser/parser.scala

```diff
@@ -0,0 +1,200 @@
+package asciibot.parser
+
+import java.io.BufferedReader
+
+import scala.collection.mutable.MutableList
+import picolib.maze.Maze
+import picolib.semantics._
+
+class AsciibotError() extends RuntimeException
+case class CharBlockError(m: String, sc: Int, ec: Int) extends AsciibotError
+case class BlockError(m: String, startChar: Int, endChar: Int,
+                      startLine: Int, endLine: Int) extends AsciibotError
+case class LineError(m: String, line: Int) extends AsciibotError
+case class FileError(m: String) extends AsciibotError
+
+object AsciibotParser {
+
+  def parseLines(program: Iterator[String]) : List[Rule] = {
+    val lines: MutableList[String]= new MutableList
+    val rules: MutableList[Rule] = new MutableList
+    program.zipWithIndex foreach { case (line, lineNum) => {
+      val noComment = line.split("//")(0)
+      if (noComment.trim.equals("")) {
+        if (lines.length > 0)
+          throw new LineError("Blank line in middle of rule", lineNum)
+      } else {
+        lines += noComment
+        if (lines.length == 3) {
```

Comment:
I feel like this is one of those cases where a constant instead of a magic number is nice